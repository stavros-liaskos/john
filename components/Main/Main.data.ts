import { MainProps } from './Main.types';
import { listData } from '../List/List.data';

const props: MainProps = {
  i18n: { todo: 'toso' },
  defaultList: listData,
};

export default props;
