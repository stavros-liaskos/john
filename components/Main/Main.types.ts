import { ListEl } from '../List/List.types';

export interface MainI18n {
  todo: string;
}

export interface MainProps {
  i18n: MainI18n;
  className: string;
  defaultList?: ListEl[];
}
