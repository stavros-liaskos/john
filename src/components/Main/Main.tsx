import FollowedArtistList from '../FollowedArtistList/FollowedArtistList';
import React, { Suspense } from 'react';
import Search from '../Search/Search';
import { useUser } from '@auth0/nextjs-auth0/client';
import Login from '../Login/Login';
import Scrapers from '../Scrape/Scrapers';
import Recommendations from '../Recommendations/Recommendations';
import Loading from '../Loading/Loading';
import { searchI18n, followedArtistListI18n, loginI18n, recommendationsI18n } from '../../i18n';

const Main: React.FunctionComponent = () => {
  const { user } = useUser();

  function loginBE() {
    window.location.assign('https://api.releaseraccoon.online/me');
  }

  function fetchy() {
    fetch('https://api.releaseraccoon.online/me', {
      headers: { 'X-Requested-With': 'JavaScript' },
    });
  }

  return (
    <main className="rr-column flex-auto">
      <div className="flex flex-col flex-auto w-full lg:w-9/12">
        <button className={'rr-text btn'} onClick={loginBE}>
          login with assign
        </button>

        <button className={'rr-text btn'} onClick={fetchy}>
          login with fetch
        </button>

        <a href="https://api.releaseraccoon.online/me">
          <button className={'rr-text btn'}>login with anchor</button>
        </a>

        {user ? (
          <>
            <Search i18n={searchI18n} />
            <Scrapers />
            <div className="flex flex-auto flex-col h24">
              <FollowedArtistList i18n={followedArtistListI18n} />
              <Suspense fallback={<Loading />}>
                <Recommendations i18n={recommendationsI18n} />
              </Suspense>
            </div>
          </>
        ) : (
          <Login i18n={loginI18n} />
        )}
      </div>
    </main>
  );
};
Main.whyDidYouRender = false;
export default Main;
