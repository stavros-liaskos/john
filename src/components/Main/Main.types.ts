export interface MainI18n {
  todo: string;
}

export interface MainProps {
  i18n: MainI18n;
}
