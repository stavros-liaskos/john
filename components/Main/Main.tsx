import { MainProps } from './Main.types';
import List from '../List/List';
import React, { useEffect, useState } from 'react';
import { ListEl } from '../List/List.types';
import Search from '../Search/Search';
import { searchI18n } from '../Search/Search.data';
import { listI18n } from '../List/List.data';
import { useUser } from '@auth0/nextjs-auth0';
import Login from '../Login/Login';
import {components} from '../../types/schema';

const Main: React.FunctionComponent<MainProps> = ({ i18n }) => {
  //const { user, error } = useUser(); // TODO use isLoading or try Suspense
  const { user, error } = { user: 'asdf',  error: null}

  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const [list, setList] = useState<components['schemas']['FollowedArtistsResponse']>();

  useEffect(() => {
    fetch(`${process.env.BE_BASE_URL}/me/followed-artists`, {
      method: 'GET',
      mode: 'cors',
      headers: {
        'Content-Type': 'application/json',
      },
      referrerPolicy: 'no-referrer',
    })
      .then(response => response.json())
      .then(data => setList(data));
  }, []);

  if (!i18n || !i18n.todo) {
    return null;
  }

  // if (isLoading) return <div>Loading...</div>;
  if (error) { // @ts-ignore
    return <div>{error.message}</div>;
  }

  return (
    <main
      className={`flex flex-col items-center justify-center w-full p-3 mb-auto min-h-[calc(100vh_-_8.5rem)]`}
      style={{ minHeight: 'calc(100vh - 8.5rem)' }} /* the tailwind class only works locally */
    >
      {user ? (
        <div className={`flex flex-col items-center justify-center w-full lg:w-9/12`}>
          <Search i18n={searchI18n} />
          <List list={list} i18n={listI18n} />
        </div>
      ) : (
        <Login
          i18n={{
            welcome: 'Welcome to Release Raccoon!',
            loginBtn: 'Register',
            text: "Receive your favorite artists' music in your email every week!",
            artistsCount: 'Artists',
            releasesCount: 'Releases',
          }}
          handleRegister={() => (window.location.href = '/api/auth/login')}
          counters={{
            artistsCounter: 4965,
            releasesCounter: 3816,
          }}
        />
      )}
    </main>
  );
};

export default Main;
