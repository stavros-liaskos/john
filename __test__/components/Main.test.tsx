import React from "react";
import { getAllByTestId, render } from "@testing-library/react";
import Main from "../../components/Main/Main";
import { mainI18n } from "../../components/Main/Main.data";

describe("Main", () => {
  it("renders without data without crashing", () => {
    // @ts-ignore
    render(<Main />);
  });

  it("renders with data", () => {
    const component = render(<Main i18n={mainI18n} />);
    const listEls = getAllByTestId(component.container, "list-el");

    expect(listEls.length).toEqual(2);
    expect(component).toMatchSnapshot();
  });
});
