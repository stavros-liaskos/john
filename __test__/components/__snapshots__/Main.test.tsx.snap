// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders with data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main>
        <h1
          class="text-3xl font-bold underline"
        >
          Powered by
           to 
          <a
            href="https://nextjs.org"
          >
            Next.js!
          </a>
        </h1>
        <div>
          <div
            data-testid="list-el"
          >
            <p>
              Led Zeppeling
            </p>
            <a
              href="www.google.com"
            >
              Led Zeppeling
            </a>
          </div>
          <div
            data-testid="list-el"
          >
            <p>
              YOYOO
            </p>
            <a
              href="www.google.com"
            >
              YOYOO
            </a>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main>
      <h1
        class="text-3xl font-bold underline"
      >
        Powered by
         to 
        <a
          href="https://nextjs.org"
        >
          Next.js!
        </a>
      </h1>
      <div>
        <div
          data-testid="list-el"
        >
          <p>
            Led Zeppeling
          </p>
          <a
            href="www.google.com"
          >
            Led Zeppeling
          </a>
        </div>
        <div
          data-testid="list-el"
        >
          <p>
            YOYOO
          </p>
          <a
            href="www.google.com"
          >
            YOYOO
          </a>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
